app/src/main/java/de/grobox/transportr/trips/detail/StopViewHolder.kt

```diff
@@ -41,8 +42,14 @@ internal class StopViewHolder(v: View, private val listener : LegClickListener)
     fun bind(stop: Stop, color: Int) {
         if (stop.arrivalTime != null) {
             setArrivalTimes(fromTime, fromDelay, stop)
+            fromTime.visibility = VISIBLE
         } else {
             fromDelay.visibility = GONE
+            if (stop.departureTime == null) {
+                // insert dummy time field for stops without times set, so that stop circles align
+                fromTime.text = getTime(context, Date())
+                fromTime.visibility = INVISIBLE
+            }
```

Comment:
I think there's still a case where a recycled view holder will be used with `fromTime` invisible: `stop.arrivalTime == null && stop.departureTime != null`.

Reply:
Okay, but when `stop.arrivalTime == null` then we actually don't need `fromTime` right? If `stop.departureTime != null` the *`toTime`* will be set some lines later and changed to be visible.

Reply:
I think `stop.departureTime` is unrelated, right?

I am concerned that even if we don't "need" `fromTime` when there's no arrival time, that its visibility is undefined in that case. It might be visible or it might not be. We would need to test both cases to see how this looks in the UI.

Reply:
Hum, the current behavior is that `fromTime` is always visible whether or not it is set to `stop.arrivalTime`. So a visible but unused `fromTime` doesn't harm (as it is now) and an invisible `fromTime` either (tested in Swiss case).

But you might be right, I think instead of hoping it will look good in the UI, I could simply add another `fromTime.visibility = VISIBLE` in a new `else` case in line 52.

Reply:
You could ;)

Reply:
Right. Would you mind testing these cases? At least in the graphic LayoutEditor?

Reply:
Sadly, it doesn't display as it should for me in the Android Studio XML preview even without changing anything. Any ideas why this is happening?

![transportr](https://user-images.githubusercontent.com/18088991/53200580-2a6bb300-3622-11e9-80b0-cfc2248367b0.png)

Reply:
I managed to test it now after `Build > Clean Project` in the XML preview with the different visibility settings. Should be ready to merge now.